src/mappings/pool.ts

```diff
@@ -117,6 +118,52 @@ function createPoolTokenEntity(id: string, pool: String, address: String): void
   poolToken.save()
 }
 
+function updatePoolLiquidity(id: string): void {
+  // let WETH = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' // mainnet
+  let WETH = '0xd0a1e359811322d97991e03f863a0c30c2cf029c' // kovan
+  let liquidity = BigDecimal.fromString('0')
+
+  let pool = Pool.load(id)
+  let tokensList: Array<Bytes> = pool.tokensList
+  if (!tokensList) return
+
+  if (tokensList.includes(Address.fromString(WETH))) {
+    let wethPoolTokenId = id.concat('-').concat(WETH)
+    let wethPoolToken = PoolToken.load(wethPoolTokenId)
+    liquidity = wethPoolToken.balance.div(wethPoolToken.denormWeight).times(pool.totalWeight)
```

Comment:
`truncate()` any BigDecimals to 18 when they can go over

Reply:
I've only truncated the final values, so the ratio for the price stay with same precision, does it make sense? Here is the commit: https://github.com/balancer-labs/balancer-subgraph/pull/33/commits/8cb02c38376e80ad56fc913301abc81db249cbec

Reply:
👍